test/unit/test-mysqlDao.js

```diff
@@ -180,6 +201,37 @@ describe('MysqlDao', function() {
         });
     });
 
+    describe('upsert', function() {
+        beforeEach(databaseSetup);
+
+        it('Should update a single row in the user_attributes table', function() {
+            return userAttrDao.upsert({
+                    user_id: 1, user_lead_id: 1, user_subscription_id: 1, site_id: 1,
```

Comment:
Good call, I'll rename the columns.  I'm going to keep the other table around though, never hurts to test against multiple sources of data.